datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -169,59 +170,26 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
             update_headers(headers, config['extra_headers'])
 
         # http://docs.python-requests.org/en/master/user/authentication/
-        auth = None
-        if config['password']:
-            if config['username']:
-                auth_type = config.get('auth_type', 'basic').lower()
-                if auth_type == 'digest':
-                    auth = requests.auth.HTTPDigestAuth(config['username'], config['password'])
-                else:
-                    if auth_type != 'basic':
-                        self.logger.debug('auth_type %s is not supported, defaulting to basic', auth_type)
-                    auth = (config['username'], config['password'])
+        auth_type = config['auth_type'].lower()
+        if auth_type not in AUTH_TYPES:
+            self.logger.warning('auth_type %s is not supported, defaulting to basic', auth_type)
+            auth_type = 'basic'
 
-            elif config['ntlm_domain']:
-                ensure_ntlm()
-
-                auth = requests_ntlm.HttpNtlmAuth(config['ntlm_domain'], config['password'])
-
-        if auth is None:
+        if auth_type == 'basic':
             if config['kerberos_auth']:
-                ensure_kerberos()
-
-                # For convenience
-                if is_affirmative(config['kerberos_auth']):
-                    config['kerberos_auth'] = 'required'
-
-                if config['kerberos_auth'] not in KERBEROS_STRATEGIES:
-                    raise ConfigurationError(
-                        'Invalid Kerberos strategy `{}`, must be one of: {}'.format(
-                            config['kerberos_auth'], ' | '.join(KERBEROS_STRATEGIES)
-                        )
-                    )
-
-                auth = requests_kerberos.HTTPKerberosAuth(
-                    mutual_authentication=KERBEROS_STRATEGIES[config['kerberos_auth']],
-                    delegate=is_affirmative(config['kerberos_delegate']),
-                    force_preemptive=is_affirmative(config['kerberos_force_initiate']),
-                    hostname_override=config['kerberos_hostname'],
-                    principal=config['kerberos_principal'],
+                self.logger.warning(
+                    'The ability to use Kerberos auth without explicitly setting auth_type to '
+                    '`kerberos` is deprecated and will be removed in Agent 7.20.0'
```

Comment:
```suggestion
                    '`kerberos` is deprecated and will be removed in Agent 8.0.0'
```

This is breaking change. Should be removed for Agent 8?

Maybe we should keep it even longer (Agent 9.0), it's quite a serious breaking change that impacts **all** integrations using the RequestsWrapper integrations.

Reply:
I'd agree on at least Agent 8 as well (but adding the deprecation here sounds fine).